app/services/api/v3/actors/top_nodes_summary.rb

```diff
@@ -84,9 +84,20 @@ def nodes_by_year_summary_for_indicator(
         )
           initialize_top_nodes(node_type, attribute)
 
+          profile_type = Api::V3::Profile.
+            joins(context_node_type: :node_type).
+            where('node_types.name' => node_type).
+            where('context_node_types.context' => @context).
+            select('profiles.name').
+            first
+
+          profile_type_name = profile_type.name unless profile_type.nil?
+
           lines = @top_nodes.map do |node|
             {
               name: node['name'],
+              node_id: node['node_id'],
+              profile_type: profile_type_name,
```

Comment:
I believe it's overkill to include it with every "line", as the profile type is going to be the same for all lines; maybe consider moving it up to the same level as "unit" in the response.